addons/binding/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/handler/ApiserviceHandler.java

```diff
@@ -0,0 +1,179 @@
+                }
+            }
+        }, 1, 1, TimeUnit.DAYS);
+        logger.debug("Refresh job scheduled to run every {} days. for '{}'", 1, getThing().getUID());
+    }
+
+    @Override
+    public void updateStatus(ThingStatus status) {
+        updateStatus(status, ThingStatusDetail.NONE, null);
+    }
+
+    /**
+     * Request auth token with read or write access.
+     * (Write access is prepared for a possible later usage for updating meters.)
+     *
+     * @param user     The username to use
+     * @param password The corresponding password
+     * @param scope    The granted scope on the api for the binding
+     */
+    private void obtainAuthTokenAndExpiryDate(String user, String password, String scope) {
+        try {
+
+            String url = new StringBuilder("https://pixometer.io/api/v1/access-token/").toString();
+            Properties urlHeader = (Properties) new Properties().put("CONTENT-TYPE", "application/json");
+
+            JsonObject httpBody = new JsonObject();
+            httpBody.addProperty("username", user);
+            httpBody.addProperty("password", password);
+            httpBody.addProperty("scope", scope);
+
+            InputStream content = new ByteArrayInputStream(httpBody.toString().getBytes(StandardCharsets.UTF_8));
+            String urlResponse = HttpUtil.executeUrl("POST", url, urlHeader, content, "application/json", 2000);
+            JsonObject responseJson = (JsonObject) new JsonParser().parse(urlResponse);
+
+            if (responseJson.has(CONFIG_BRIDGE_AUTH_TOKEN)) {
+
+                // Store the expire date for automatic token refresh
+                Integer expiresIn = Integer.parseInt(responseJson.get("expires_in").toString());
+                setTokenExpiryDate(ZonedDateTime.now().plusSeconds(expiresIn));
+
+                setAuthToken(responseJson.get(CONFIG_BRIDGE_AUTH_TOKEN).toString().replaceAll("\"", ""));
+                updateStatus(ThingStatus.ONLINE);
+                return;
+            }
+
+            String errorMsg = new StringBuilder("Invalid Api Response ( ").append(responseJson.toString()).append(" )")
+                    .toString();
+
+            throw new Exception(errorMsg);
+
+        } catch (Exception e) {
+            String errorMsg = new StringBuilder("Could not obtain auth token.").append(" ")
+                    .append("Please check your configured account credentials.")
+                    .append(System.getProperty("line.separator")).append(this.getThing().getUID())
+                    .append(System.getProperty("line.separator")).append(e.getMessage()).toString();
+
+            logger.error(errorMsg, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorMsg);
+        }
+    }
+
+    /**
+     * Getters and Setters
+     */
+
+    public String getAuthToken() {
+        return authToken;
+    }
+
+    private void setAuthToken(String authToken) {
+        this.authToken = authToken;
```

Comment:
The authToken is only stored in memory. If the binding or openHAB is reloaded it will be gone. If you want to persist the value you can use `editConfiguration` and `updateConfiguration` methods.

Reply:
It's probably used on several places. But the one that comes to mind, and which I'm working on, and although not yet a merged binding, it contains both usages: the Spotify binding: https://github.com/openhab/openhab2-addons/pull/2295/files

Reply:
Here is an other example of `editProperties`: https://github.com/openhab/openhab2-addons/blob/4e13f8723b967ddfa3622beb776c91ef9f02628a/addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java#L191